app/src/main/java/nu/yona/app/api/manager/impl/ActivityCategoryManagerImpl.java

```diff
@@ -52,33 +53,13 @@ public ActivityCategoryManagerImpl(Context context) {
      * @param listener
      */
     @Override
-    public void getActivityCategoriesById(final DataLoadListener listener) {
-        try {
-            User user = YonaApplication.getEventChangeManager().getDataState().getUser();
-            if (user != null && user.getLinks() != null && user.getLinks().getYonaAppActivity() != null && !TextUtils.isEmpty(user.getLinks().getYonaAppActivity().getHref())) {
-                activityCategoriesNetwork.getActivityCategories(authenticateDao.getUser().getLinks().getYonaAppActivity().getHref(), new DataLoadListener() {
-                    @Override
-                    public void onDataLoad(Object result) {
-                        updateActivityCategories(result, null);
-                        if (listener != null) {
-                            listener.onDataLoad(result);
-                        }
-                    }
-
-                    @Override
-                    public void onError(Object errorMessage) {
-                        if (listener != null) {
-                            listener.onError(errorMessage);
-                        }
-                    }
-                });
-            } else {
-                if (listener != null) {
-                    listener.onError(new ErrorMessage(mContext.getString(R.string.urlnotfound)));
-                }
-            }
-        } catch (Exception e) {
-            AppUtils.throwException(ActivityCategoryManagerImpl.class.getSimpleName(), e, Thread.currentThread(), listener);
+    public void getActivityCategoriesById(DataLoadListener listener) {
+        User user = YonaApplication.getEventChangeManager().getDataState().getUser();
+        if (user != null && user.getLinks() != null && user.getLinks().getYonaAppActivity() != null && !TextUtils.isEmpty(user.getLinks().getYonaAppActivity().getHref())) {
+            DataLoadListenerImpl listenerWrapper = new DataLoadListenerImpl((result) -> updateActivityCategories(result, null), listener);
+            activityCategoriesNetwork.getActivityCategories(authenticateDao.getUser().getLinks().getYonaAppActivity().getHref(), listenerWrapper);
```

Comment:
What happens here? Why are you calling ``getActivityCategories`` with the app activity reporting URL? These seem unrelated to me.

Reply:
Will debug this and see what is happening here. But it works even though they are not related!

Reply:
Great, as this is concerning me quite a bit.

Reply:
URL specified in the method is ignored because of the @GET annotation. This is appending the path 'activityCategories/' to the base url of the network class. So it is working without any issues. I will remove the url argument
'''
@GET(ApiList.ACTIVITY_CATEGORIES)
    Call<ActivityCategories> getActivityCategories(@Header(NetworkConstant.ACCEPT_LAUNGUAGE) String acceptLanaguage);
'''

Reply:
Right! That explains why it didn't fail.